src/main/java/com/facebook/openwifirrm/ucentral/models/AggregatedState.java

```diff
@@ -0,0 +1,234 @@
+/*
+ * Copyright (c) Meta Platforms, Inc. and affiliates.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+package com.facebook.openwifirrm.ucentral.models;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+import com.facebook.openwifirrm.ucentral.models.State.Interface.SSID.Association;
+import com.facebook.openwifirrm.ucentral.models.State.Interface.SSID.Association.Rate;
+
+/**
+ * Aggregation model for State aggregation. Only contains info useful for analysis.
+ */
+public class AggregatedState {
+
+	public static class AggregatedRate {
+		public long bitRate;
+		public int chWidth;
+		public List<Integer> mcs;
+
+		/** Constructor with no args */
+		public AggregatedRate() {
+			mcs = new ArrayList<>();
+		}
+
+		/** Add a Rate to the AggregatedRate */
+		public void add(Rate rate) {
+			if (rate == null) {
+				return;
+			}
+			if (mcs == null || mcs.isEmpty()) {
+				bitRate = rate.bitrate;
+				chWidth = rate.chwidth;
+			}
+			mcs.add(rate.mcs);
+		}
+
+		/** Add an AggregatedRate with the same channel_width to the AggregatedRate */
+		public void add(AggregatedRate rate) {
+			if (rate.chWidth != chWidth) {
+				return;
+			}
+			if (mcs == null || mcs.isEmpty()) {
+				bitRate = rate.bitRate;
+				chWidth = rate.chWidth;
+			}
+			mcs.addAll(rate.mcs);
+		}
+	}
+
+	public String bssid;
+	public String station;
+	public long connected;
+	public long inactive;
+	public List<Integer> rssi;
+	public long rxBytes;
+	public long rxPackets;
+	public AggregatedRate rxRate;
+	public long txBytes;
+	public long txDuration;
+	public long txFailed;
+	public long txPackets;
+	public AggregatedRate txRate;
+	public long txRetries;
+	public int ackSignal;
+	public int ackSignalAvg;
+	public Radio radio;
+
+	public static class Radio {
+		public int channel;
+		public int channelWidth;
+		public int txPower;
+
+		public Radio() {}
+
+		public Radio(int channel, int channelWidth, int txPower) {
+			this.channel = channel;
+			this.channelWidth = channelWidth;
+			this.txPower = txPower;
+		}
+
+		public Radio(Map<String, Integer> radioInfo) {
+			channel = radioInfo.getOrDefault("channel", null);
```

Comment:
You cannot assign null to primitive types like int. Please either box it or use -1 (or other feasible "invalid" number) as default. Same for `channelWidth` and `txPower`